force_bdss/mco/optimizers/scipy_optimizer.py

```diff
@@ -0,0 +1,35 @@
+from typing import Callable
+from traits.api import (
+    Enum,
+    provides,
+    HasStrictTraits
+)
+
+from force_bdss.mco.optimizers.i_optimizer import IOptimizer
+
+from scipy import optimize as scipy_optimize
+
+
+@provides(IOptimizer)
+class ScipyOptimizer(HasStrictTraits):
+    """ Optimization of a scalar function using scipy.
+    """
+
+    #: Algorithms available to work with
+    algorithms = Enum("SLSQP", "Nelder-Mead", "Powell", "CG", "BFGS",
+                      "Newton-CG", "L-BFGS-B", "TNC", "COBYLA",
+                      "trust-constr", "dogleg",
+                      "trust-ncg", "trust-exact", "trust-krylov")
+
+    def optimize_function(self, func: Callable, x0: list, bounds: list):
```

Comment:
Can we put the parameter types in a docstring here, rather than in the signature?

Reply:
yeah